client/transport-netty/src/main/java/org/elasticsearch/transport/client/PreBuiltTransportClient.java

```diff
@@ -50,14 +66,16 @@ public PreBuiltTransportClient(Settings settings, Collection<Class<? extends Plu
         super(settings, Settings.EMPTY, addPlugins(plugins, PRE_INSTALLED_PLUGINS));
     }
 
-    /**
-     * The default transport implementation for the transport client.
-     */
-    public static final class TransportPlugin extends Netty3Plugin {
-        // disable assertions for permissions since we might not have the permissions here
-        // compared to if we are loaded as a real module to the es server
-        public TransportPlugin(Settings settings) {
-            super(Settings.builder().put("netty.assert.buglevel", false).put(settings).build());
```

Comment:
the reason why this was set by default was that the users don't need to set it, now they do. I wonder if it should be set as additional settings then instead.